concepts/0519-goal-codes/README.md

```diff
@@ -0,0 +1,149 @@
+# 0519: Goal Codes
+- Authors: [Daniel Hardman](daniel.hardman@gmail.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-07-24
+- Status Note: surfaces thinking first used in [0434 OOB Protocol](../../features/0434-outofband/README.md) and in [Co-protocols](https://docs.google.com/presentation/d/17hk6QqLW5M9E4TBPZwXIUBu9eEinMNXReceyZTF4LpA/edit)
+- Start Date: 2020-01-01
+- Tags: concept
+
+## Summary
+
+Explain how different parties in an SSI ecosystem can communicate about their intentions in a way that is understandable by humans and by automated software.
+
+## Motivation
+
+[Agents](../0004-agents/README.md) exist to achieve the intents of their owners. Those intents largely unfold through [protocols](../0003-protocols/README.md). Sometimes intelligent action in these protocols depends on a party declaring their intent. We need a standard way to do that.
+
+## Tutorial
+
+Our early learnings in SSI focused on VC-based proving with a very loose, casual approach to context. We did demos where Alice connects with a potential employer, Acme Corp -- and we assumed that each of the interacting parties had a shared understanding of one another's needs and purposes.
+
+But in a mature SSI ecosystem, where unknown agents can contact one another for arbitrary reasons, this context is not always easy to deduce. Acme Corp's agent may support many different protocols, and Alice may interact with Acme in the capacity of customer or potential employee or vendor. Although we have [feature discovery](../../features/0031-discover-features/README.md) to learn what's possible, and we have [machine-readable governance frameworks](../0430-machine-readable-governance-frameworks/README.md) to tell us what rules might apply in a given context, we haven't had a way to establish the context in the first place. When Alice contacts Acme, a context is needed before a governance framework is selectable, and before we know which features are desirable.
+
+The key ingredient in context is __intent__. If Alice says to Acme, "I'd like to connect,", Acme wants to be able to trigger different behavior depending on whether Alice's intent is to be a customer, apply for a job, or audit Acme's taxes. This is the purpose of a goal code.
```

Comment:
Have you given any thought to combining this concept with Action Menu to achieve the desired goal? It seems like the power of Action Menu (corresponding to the simple power that was the generation 1 HTML `form` element) would merge nicely with goal-code to enable the machine handling of Action Menu forms.

Reply:
@swcurran in that case I'd propose modifying `action-menu` such that the `menu` message can be sent in response to *any* didcomm message although that would add complexity to the state machines...

I'd rather keep having the initiating party start with the goal code, more so when both are operating under a governance framework.